docs/schulcloud-server/Coding-Guidelines/modules-submodules.md

```diff
@@ -0,0 +1,66 @@
+# Implementation and usage of modules, submodule and barrel files in our project
+
+In this guide, we'll cover how to use modules, submodules and barrel files in our project. These concepts help you organize your code into separate files and directories, making it easier to manage and maintain.
+
+## Modules and Submodules
+
+In our project, modules are a way to create separate scopes. This means that interfaces, use cases, services, etc., declared in a module are not visible outside the module unless they are explicitly exported using the `export` keyword. To import a module, you use the `import` keyword followed by the module name. Here's an example:
+
+```typescript
+import { ModuleName } from '@modules/module-name';
+```
+
+Submodules are modules that are part of a larger module. They should be used within the main module. If it is necessary to use parts of the submodule outside the main module, the main module should export this via its barrel file(index.ts):
+
+
+```typescript
+// @modules/module-name/index.ts
+export { SubmoduleServiceName } from './submodule-name/service.ts';
```

Comment:
Ich denke es ist besser wenn wir hier darauf Hinweisen das submodule auch eigene in sich geschlossene Module sind die wiederum über ihr barrel file angesprochen werden. 

Ich überlege gerade ob ich nicht auch ein kleines Bild von unseren Modulen male, bzw. für submodule. 

An sich ist die Kernaussage von dir das wir über Ordner Scopes festlegen auch für die Layer gültig. Daher nicht nur vertikal, sondern horizontal.

Reply:
Bild gemalt und gepushed.